File: day-01/part-1/ayoub.py
from submission import Submission


class AyoubSubmission(Submission):

	def run(self, s):
		s = list(map(int, list(s)))
		n = len(s)
		r = 0
		for i in range(n):
			if s[i] == s[(i+1) % n]:
				r += s[i]
		return r

File: day-13/part-2/ayoub.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func run(s string) string {
	lines := strings.Split(s, "\n")
	for x := 0; ; x++ {
		caught := false
		for _, line := range lines {
			parts := strings.Split(line, ": ")
			idx, _ := strconv.Atoi(parts[0])
			depth, _ := strconv.Atoi(parts[1])
			if (idx+x)%((depth-1)*2) == 0 {
				caught = true
				break
			}
		}
		if !caught {
			return fmt.Sprintf("%d", x)
		}
	}
}

func main() {
	fmt.Println(run(os.Args[1]))
}
